acceptance/openstack/identity/v3/roles_test.go

```diff
@@ -41,21 +41,63 @@ func TestRolesGet(t *testing.T) {
 		t.Fatalf("Unable to obtain an identity client: %v", err)
 	}
 
-	allPages, err := roles.List(client, nil).AllPages()
+	role, err := FindRole(t, client)
 	if err != nil {
-		t.Fatalf("Unable to list roles: %v", err)
-	}
-
-	allRoles, err := roles.ExtractRoles(allPages)
-	if err != nil {
-		t.Fatalf("Unable to extract roles: %v", err)
+		t.Fatalf("Unable to find a role: %v", err)
 	}
 
-	role := allRoles[0]
 	p, err := roles.Get(client, role.ID).Extract()
 	if err != nil {
 		t.Fatalf("Unable to get role: %v", err)
 	}
 
 	tools.PrintResource(t, p)
 }
+
+func TestAssignToUserOnProject(t *testing.T) {
+	client, err := clients.NewIdentityV3Client()
+	if err != nil {
+		t.Fatalf("Unable to obtain an indentity client: %v", err)
+	}
+
+	project, err := FindProject(t, client)
```

Comment:
It'd be better to create a temporary project using the `CreateProject(t, client, nil)` function. This localizes all testing to a disposable project.